AppCenterCrashes/AppCenterCrashes/MSCrashes.mm

```diff
@@ -277,6 +279,22 @@ - (instancetype)init {
      * Setting this flag will let application crash on uncaught exceptions.
      */
     [[NSUserDefaults standardUserDefaults] registerDefaults:@{ @"NSApplicationCrashOnExceptions" : @YES }];
+
+    /*
+     * Exceptions on the main thread of a Cocoa application do not typically rise to the level of
+     * the uncaught exception handler because the global application object catches all such exceptions.
+     * See: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Exceptions/Concepts/UncaughtExceptions.html
+     */
+    if (![MSAppCenter isDebuggerAttached]) {
+      SEL selector = @selector(reportException:);
+      Method method = class_getInstanceMethod([NSApplication class], selector);
+      IMP implementation = class_getMethodImplementation([MSCrashes class], selector);
+      if (method && implementation) {
```

Comment:
I think this needs broader discussion. Personally, I think this is a great feature - this was planned in HockeyApp, see https://github.com/bitstadium/HockeySDK-Mac/blob/6a6e31c3cafd4991672bcbaf1a48e135a878ae5b/Classes/CrashReporting/BITCrashExceptionApplication.h - but we want to be cautious about the swizzling. I'd like to see this as an opt-in API or at least have a discussion around it. Also: have we looked into side-effects of this thoroughly?

Reply:
As discussed, we want an opt-out for the swizzling.